### PR TITLE
Update dialect with missing type registrations.

### PR DESCRIPTION
**Pull request by tjoneslo**

### CODE REVIEW

File: src/main/java/org/hibernate/dialect/FoundationDBSQLDialect.java [hunk 1]
```diff
@@ -47,6 +47,8 @@ public FoundationDBSQLDialect () {
         registerColumnType( Types.LONGVARBINARY, "varchar($l) for bit data" );
         registerColumnType( Types.BINARY, 1024, "char($l) for bit data");
         registerColumnType( Types.BINARY, "char(1024) for bit data");
+        registerColumnType( Types.NUMERIC, "decimal($p,$s)");
+        registerColumnType (Types.LONGVARCHAR, "longtext");
```
Comment (nathanlws) [inline src/main/java/org/hibernate/dialect/FoundationDBSQLDialect.java]: CLOB and LONGTEXT both map to the same type right now but CLOB probably has a better chance of surviving types clean up.

**Comment by nathanlws** [comment]
Did you notice my comment here @tjoneslo ?

**Comment by tjoneslo** [comment]
I did, then promptly got distracted by another thing. I'll look again today.

**Comment by nathanlws** [comment]
Approve

**Pull request merged**